go-controller/pkg/ovn/pods.go

```diff
@@ -414,6 +430,22 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			pod.Namespace, pod.Name, time.Since(start), libovsdbExecuteTime, podAnnoTime)
 	}()
 
+	// it is possible to try to add a pod here that has no node. For example if a pod was deleted with
+	// a finalizer, and then the node was removed. In this case the pod will still exist in a running state.
+	// Terminating pods should still have network connectivity for pre-stop hooks or termination grace period
+	if _, err := oc.watchFactory.GetNode(pod.Spec.NodeName); kerrors.IsNotFound(err) &&
+		oc.lsManager.GetSwitchSubnets(pod.Spec.NodeName) == nil {
+		if util.PodTerminating(pod) {
+			klog.Infof("[%s/%s] Ignoring adding networking for terminating pod on deleted node: %s",
+				pod.Namespace, pod.Name, pod.Spec.NodeName)
+			return nil
+		} else {
+			// unknown condition how we are getting a non-terminating pod without a node here
+			klog.Errorf("[%s/%s] Non-existent node in API for pod with unknown "+
+				"condition: %s", pod.Namespace, pod.Name, pod.Spec.NodeName)
+		}
+	}
```

Comment:
I checked that the add node code path adds the pods as well. But I cannot avoid having this unconscious feeling that we should trigger a retry here as well.

Reply:
good point, we could retry here in the addLogicalPort path. I don't think it will hurt anything. Should we add it in both cases (terminating as well as the unknown path)?

Reply:
I would add it for both cases.
The terminating pod would eventually go away if the node was deleted? And if the node was not deleted, then it's good that we retry event if we might already be or would be retrying because of the node add code path.
The unknown state, being unknown, might be worth a retry as well in case it's a state that resorts itself to some other workable state for the pod.
Since we are not aware of any harm that we might cause, I would just do it.

Reply:
> resorts itself to some other workable state for the pod

for which we would get an update event I guess. Anyway, I would still do it, specially if that means getting rid or specific code to handle specific errors.

Reply:
I'm still going to return specific errors because the pod terminating is the only condition where we know it is possible. I want to differentiate that from an unknown case.